tests/unittests/tiglTanks.cpp

```diff
@@ -0,0 +1,411 @@
+/*
+ * Copyright (C) 2007-2022 German Aerospace Center (DLR/SC)
+ *
+ * Created: 2023-12-29 Marko Alder <marko.alder@dlr.de>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * @file
+ * @brief Tests for testing duct functions.
+ */
+
+#include "test.h"
+#include "tigl.h"
+
+#include "CCPACSFuelTank.h"
+#include "CCPACSFuelTanks.h"
+#include "CCPACSConfigurationManager.h"
+#include "CTiglUIDManager.h"
+
+#include "CCPACSVessel.h"
+#include "CNamedShape.h"
+
+namespace
+{
+// Error message constants for exception tests
+constexpr const char* tankTypeExceptionString =
+    "This method is only available for vessels with segments. No segment found.";
+constexpr const char* invalidIndexMessage    = "Invalid index in CCPACSFuselageSections::GetSection";
+constexpr const char* wrongSectionUIDMessage = "GetSectionFace: Could not find a fuselage section for the given UID";
+
+// Helper function to check exception messages.
+void CheckExceptionMessage(const std::function<void()>& func, const char* expectedMessage)

```

Comment:
nice one!

Reply:
Just one thought: Why `const char*` and not `std::string(_view)`?

Reply:
Yup, switched to `std::string_view` and moved to testUtils (b2931900)